src/gas-price/gas-price.test.ts

```diff
@@ -373,12 +365,22 @@ describe(getRecommendedGasPrice.name, () => {
   });
 });
 
-describe(fetchAndStoreGasPrice.name, () => {
+describe(gasPriceModule.getGasPrice.name, () => {
+  it('returns gas price as bigint when internal rpc call returns a valid hex string', async () => {
```

Comment:
Can you add one more test when the provider fails? I checked [Ethereum RPC spec](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_gasprice) (and some other sources) and I think we can rely that when the API returns something it will be a hex string...